Source/Core/Core/PowerPC/Jit64/JitAsm.cpp

```diff
@@ -113,15 +113,21 @@ void Jit64AsmRoutineManager::Generate()
   const bool assembly_dispatcher = true;
   if (assembly_dispatcher)
   {
-    if (m_jit.GetBlockCache()->GetFastBlockMap())
+    if (m_jit.GetBlockCache()->GetEntryPoints())
     {
-      u64 icache = reinterpret_cast<u64>(m_jit.GetBlockCache()->GetFastBlockMap());
-      MOV(32, R(RSCRATCH), PPCSTATE(pc));
+      MOV(32, R(RSCRATCH2), PPCSTATE(msr));
+      AND(32, R(RSCRATCH2), Imm32(JitBaseBlockCache::JIT_CACHE_MSR_MASK));
+      SHL(64, R(RSCRATCH2), Imm8(28));
+
+      MOV(32, R(RSCRATCH_EXTRA), PPCSTATE(pc));
+      OR(64, R(RSCRATCH_EXTRA), R(RSCRATCH2));
 
+      u64 icache = reinterpret_cast<u64>(m_jit.GetBlockCache()->GetEntryPoints());
       MOV(64, R(RSCRATCH2), Imm64(icache));
-      // Each 4-byte offset of the PC register corresponds to a 8-byte offset
-      // in the lookup table due to host pointers being 8-bytes long.
-      MOV(64, R(RSCRATCH), MComplex(RSCRATCH2, RSCRATCH, SCALE_2, 0));
+      // The entry points map is indexed by ((msrBits << 26) | (address >> 2) << 3)
+      // The 3 corresponds to an 8 byte 64-bit pointer and that means we need to shift
+      // msr left by 29 bits and address left by 1 bit to get the index to the map.
```

Comment:
I think it would be slightly more accurate to say:

```suggestion
      // The entry points map is indexed by ((msrBits << 26) | (address >> 2) << 3)
      // The 3 corresponds to an 8 byte 64-bit pointer and that means we need to shift
      // msr left by 29 bits and address left by 1 bit to get the offset in the map.
```

The index is `(msrBits << 26) | (address >> 2)`, without shifting left by three.

Reply:
Technically the index is ((msrBits & JitBaseBlockCache::JIT_CACHE_MSR_MASK) << 26) | (address >> 2), but not sure which is clearer to read.

Reply:
Yeah, let's skip the masking in the comment. It just makes it hard to read.